### PR TITLE
filter: add `过滤 TiDB parser 不支持的 SQL 语句`

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

- add doc for `过滤 TiDB parser 不支持的 SQL 语句`

### What is changed and how it works?

- add some desc for `过滤 TiDB parser 不支持的 SQL 语句`

### Check List 

Tests 

 - No code

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@GregoryIan @amyangfei PTAL

File: docs/dm/zh_CN/features/binlog-filter.md [hunk 1]
```diff
@@ -123,4 +124,22 @@ filters:
...
+
+注意：全局过滤规则必须设置尽可能严格以避免不预期地过滤需要同步的数据
+
+设置下面的规则过滤 TiDB parser 不支持的 `PARTITION statement`
+
+```yaml
+filters:
+  filter-partition-rule:
+    schema-pattern: "*"
+    sql-pattern: ["ALTER\\s+TABLE[\\s\\S]*ADD\\s+PARTITION", "ALTER\\s+TABLE[\\s\\S]*DROP\\s+PARTITION"]
```
Comment (kennytm) [inline docs/dm/zh_CN/features/binlog-filter.md:143]: If the YAML parser works correctly, you can avoid these double backslashes by using [single quote](https://yaml.org/spec/1.2/spec.html#id2776092).

```suggestion
    sql-pattern: ['ALTER\s+TABLE[\s\S]*ADD\s+PARTITION', 'ALTER\s+TABLE[\s\S]*DROP\s+PARTITION']
```

**Comment by IANTHEREAL** [comment]
LGTM

File: docs/dm/zh_CN/features/binlog-filter.md (hunk 1 above)
Comment (csuzhangxc) [reply docs/dm/zh_CN/features/binlog-filter.md:143]: :+1: yeah, I tested it and our YAML parser (*gopkg.in/yaml.v2*) works correctly.
but our users maybe not familiar with YAML, and can not differ single quote and double quote well, so I still want to keep them uniform.

**Comment by csuzhangxc** [comment]
@kennytm PTAL again

**Review by kennytm — APPROVED** [review]
LGTM

**Pull request merged**